### ISSUE TITLE
Does __fmaf really help?

### ISSUE THREAD

**Issue by kwea123** [MrNeRF/gaussian-splatting-cuda#36]
I see you replace basically every operation with this intrinsics, but according to [this](https://stackoverflow.com/questions/33796499/cuda-intrinsic-function-advantage) there is zero speedup, they serve only for predicatable rounding

**Comment by MrNeRF** [comment]
Yes, I've had the same experience. I tried it mainly because the NVIDIA CUDA profiler kept suggesting these operations to enhance speed. However, in terms of speed, it doesn't seem to be beneficial at all. Instead, it tends to obfuscate and makes the code more difficult to understand.
